@here/harp-materials/lib/SolidLineMaterial.ts

```diff
@@ -511,28 +556,32 @@ export class SolidLineMaterial extends THREE.RawShaderMaterial
         return this.uniforms.diffuse.value as THREE.Color;
     }
     set color(value: THREE.Color) {
-        this.uniforms.diffuse.value = value;
+        this.uniforms.diffuse.value.copy(value);
     }
 
     /**
      * Outline color.
+     *
+     * @note The width of outline ([[outlineWidth]]) need to be also set to enable outlining.
      */
     get outlineColor(): THREE.Color {
         return this.uniforms.outlineColor.value as THREE.Color;
     }
     set outlineColor(value: THREE.Color) {
-        this.uniforms.outlineColor.value = value;
+        this.uniforms.outlineColor.value.copy(value);
     }
 
     /**
      * Dash color.
+     *
+     * @note The property [[gapSize]] need to be set to enable dashed line.
      */
     get dashColor(): THREE.Color {
         return this.uniforms.dashColor.value as THREE.Color;
     }
     set dashColor(value: THREE.Color) {
-        this.uniforms.dashColor.value = value;
-        this.defines.USE_DASH_COLOR = 1.0;
+        this.uniforms.dashColor.value.copy(value);
+        setShaderMaterialDefine(this, "USE_DASH_COLOR", 1.0);
```

Comment:
Does this need to be a numeric define? Seems like it could also be implemented with a "boolean" define similar to `USE_COLOR`, `USE_OUTLINE`.

Reply:
Of course! I left is so to preserve entire shader code compatibility...
I did not wanted to make entire "world" better, but limit change to minimum, while in some places (see CAPS_MODE) I could not hold myself from cleaning up :). Probably you clearly see why I have done it.